### ISSUE TITLE
[active bug] Setup of sensor platform fails

### ISSUE THREAD

**Issue by CantBeCorrect** [Chouffy/home_assistant_tgtg#30]
First of all: Thanks for this great integration!
Since yesterday, the tgtg platform can't setup sensors for me anymore. The stack trace indicates the API does require to solve a captcha, which can't be served obviously.

My first instinct would be the API is applying a rate limit, but I have an update_interval of 90 seconds, which imo is far from enough to warrant rate limiting. 

So my second guess is TGTG changed their API and the integration needs to be updated, hence this issue. 

Any help is much appreciated! 

Stack trace:
```
Error while setting up tgtg platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/tgtg/sensor.py", line 67, in setup_platform
    add_entities([TGTGSensor(each_item_id)])
  File "/config/custom_components/tgtg/sensor.py", line 87, in __init__
    self.update()
  File "/config/custom_components/tgtg/sensor.py", line 180, in update
    self.tgtg_answer = tgtg_client.get_item(item_id=self.item_id)
  File "/usr/local/lib/python3.9/site-packages/tgtg/__init__.py", line 244, in get_item
    self.login()
  File "/usr/local/lib/python3.9/site-packages/tgtg/__init__.py", line 121, in login
    self._refresh_token()
  File "/usr/local/lib/python3.9/site-packages/tgtg/__init__.py", line 111, in _refresh_token
    raise TgtgAPIError(response.status_code, response.content)
tgtg.exceptions.TgtgAPIError: (403, b'<html><head><title>apptoogoodtogo.com</title><style>#cmsg{animation: A 1.5s;}@keyframes A{0%{opacity:0;}99%{opacity:0;}100%{opacity:1;}}</style></head><body style="margin:0"><p id="cmsg">Please enable JS and disable any ad blocker</p><script>var dd={\'cid\':\'AHrlqAAAAAMAVhFJaKj7JkABIAMA8v8p7wALcUZE2qeLnQ==\',\'hsh\':\'1D42C2CA6131C526E09F294FE96F94\',\'t\':\'fe\',\'r\':\'b\',\'s\':35558,\'e\':\'6177e810c7ac55bdd21d8debfd47fb4b86161caf13e44a73112e4d839a8b6ccd\',\'host\':\'geo.captcha-delivery.com\'}</script><script src="https://ct.captcha-delivery.com/c.js"></script></body></html>\n')

```

**Comment by Chouffy** [comment]
Hi @CantBeCorrect
Thanks for your nice words, glad that it's useful 😊
I can also reproduce the error, which is coming from the upstream module as well - [see this reply](https://github.com/ahivert/tgtg-python/issues/154#issuecomment-1155040209)

The fix is to update useragents, which today are defined in the `tgtg` library. So as soon as an update come, I'll update this module.

I want also to be able to define useragents manually in Home Assistant, so we are less reliant on `tgtg` updates.
I'll try to do it in the coming days.

**Comment by CantBeCorrect** [comment]
As a very temporary fix, I successfully added a custom user agent to the `TgtgClient`. (From [https://github.com/ahivert/tgtg-python/issues/154#issuecomment-1158059839](here)):

```
def setup_platform(
    hass: HomeAssistant,
    config: ConfigType,
    add_entities: AddEntitiesCallback,
    discovery_info: DiscoveryInfoType | None = None,
) -> None:
    """Set up the sensor platform."""

    username = config[CONF_USERNAME]
    item = config[CONF_ITEM]
    access_token = config[CONF_ACCESS_TOKEN]
    refresh_token = config[CONF_REFRESH_TOKEN]
    user_id = config[CONF_USER_ID]
    user_agent = '**TGTG/22.2.1 Dalvik/2.1.0 (Linux; U; Android 9; SM-G955F Build/PPR1.180610.011)**'

    global tgtg_client

    # Log in with tokens
    tgtg_client = TgtgClient(
        access_token=access_token, refresh_token=refresh_token, user_id=user_id, user_agent=user_agent
    )
```

**Comment by Chouffy** [comment]
I've published 4.1.0, which allow a custom user_agent to be defined in `configuration.yaml` - see [Readme](https://github.com/Chouffy/home_assistant_tgtg#configuration-option)

Also, i've [defaulted the user_agent](https://github.com/Chouffy/home_assistant_tgtg/blob/6f2604a308e2948426e30f180d4c9fec1290ec80/custom_components/tgtg/sensor.py#L40) to one that [we know works](https://github.com/ahivert/tgtg-python/issues/154#issuecomment-1158059839)

But it's not super good practice I think - let's open another issue to track that improvement
